### PR TITLE
row.get returns Some(" ") instead of returning None when allowMissingColumn flag is set

### PR DESCRIPTION
**Pull request by dsommer**
row.get returns `Some(" ")` instead of returning `None` when `allowMissingColumn` flag is set to true and the field is empty. None is correctly returned when `allowMissingColumn` is not set.
@clintval

### CODE REVIEW

File: src/main/scala/com/fulcrumgenomics/commons/util/DelimitedDataParser.scala [hunk 1]
```diff
@@ -73,7 +73,7 @@ class Row private[util] (private val headerIndices: Map[String,Int], private val
   /** Gets a value from the specified column. If the value is null or empty returns None.  If
     * `allowMissingColumn` is true, then None is returned if the given column name is not present. */
```
Comment (nh13) [inline src/main/scala/com/fulcrumgenomics/commons/util/DelimitedDataParser.scala:74]: I had to re-read this comment a few times to see what the intended behavior was.  I think will make it clearer, along with your bugfix.
```suggestion
  /** Gets a value from the specified column. If the value is null or empty returns None.  If
    * `allowMissingColumn` is true, then None is also returned if the given column name is not present. */
```

File: src/main/scala/com/fulcrumgenomics/commons/util/DelimitedDataParser.scala [hunk 2]
```diff
@@ -73,7 +73,7 @@ class Row private[util] (private val headerIndices: Map[String,Int], private val
...
   def get[A](column: String, allowMissingColumn: Boolean = false)(implicit tag: ru.TypeTag[A]): Option[A] =
-    if (allowMissingColumn) headerIndices.get(column).map(apply[A]) else get(headerIndices(column))
+    if (allowMissingColumn && ! headerIndices.contains(column)) None else get(headerIndices(column))
```
Comment (nh13) [inline src/main/scala/com/fulcrumgenomics/commons/util/DelimitedDataParser.scala]: How about this? If you go with your implementation, please change `! headerIndices` to `!headerIndices`
```suggestion
    if (allowMissingColumn) headerIndices.get(column).flatMap(get[A]) else get(headerIndices(column))
```

**Review by nh13 — APPROVED** [review]

File: src/test/scala/com/fulcrumgenomics/commons/util/DelimitedDataParserTest.scala [hunk 3]
```diff
@@ -210,4 +210,12 @@ class DelimitedDataParserTest extends UnitSpec {
     row.get[Int]("d", allowMissingColumn=true) shouldBe None
     an[Exception] shouldBe thrownBy { row.get[Int]("d", allowMissingColumn=false) }
   }
+
+  it should "test empty fields returning None when allowMissingColumn is true" in {
```
Comment (tfenne) [inline src/test/scala/com/fulcrumgenomics/commons/util/DelimitedDataParserTest.scala]: the string should describe the intended behavior, not what the test should do.  So in this case perhaps:

```scala
it should "return None for missing values even with allowMissingColumn is true"
```

File: src/main/scala/com/fulcrumgenomics/commons/util/DelimitedDataParser.scala (hunk 2 above)
Comment (tfenne) [reply src/main/scala/com/fulcrumgenomics/commons/util/DelimitedDataParser.scala]: I find both a bit hard to follow.  I think I would suggest a more verbose approach that's a little easier to read:

```scala
headerIndices.get(column) match {
  case None      => if (allowMissingColumn) None else throw new NoSuchElementException(column)
  case Some(idx) => get[A](idx)
}
```

Either way it would be nice to avoid the double lookup of the column index for the common/positive path.

**Review by tfenne — CHANGES_REQUESTED** [review]
Couple of simple suggestions.

**Comment by dsommer** [comment]
@tfenne I like the verbose match statement solution. Removed `if` statement and added another match case which seems even cleaner. h/t mike
```scala
    (headerIndices.get(column), allowMissingColumn) match {
      case (Some(idx), _) => get[A](idx)
      case (None, true)   => None
      case (None, false)  => throw new NoSuchElementException(column)
    }
```

**Review by tfenne — APPROVED** [review]
